Shepherd reverts issue labels on rebase force-push failure even when PR is already merged

## Problem

When the shepherd's rebase phase fails (specifically "rebase succeeded but force-push failed"), the shepherd exits with code 4 and runs label cleanup that reverts the issue from `loom:building` back to `loom:issue` — even when the PR has already been **merged**.

This means a successfully completed issue gets put back into the ready queue, where another shepherd could pick it up and attempt duplicate work.

## Observed During

Shepherd run for #2507 (`/shepherd 2507 -m`):

```
[06:54:09] [ERROR] rebase succeeded but force-push failed
...
[06:54:11] [INFO] Label cleanup: reverted issue #2507 from loom:building to loom:issue
[06:54:11] [OK] Released claim for issue #2507
```

At the time of this cleanup, PR #2514 was already merged:
```json
{"state": "MERGED", "labels": [{"name": "loom:pr"}]}
```

## Expected Behavior

Before reverting labels or declaring failure, the shepherd should check whether the PR associated with the issue is already merged. If merged:

1. Do **not** revert the issue label to `loom:issue`
2. Close the issue if not already closed
3. Exit with code 0 (success), not code 4
4. Log the force-push failure as a warning, not a fatal error

## Affected Code

- Shepherd rebase phase error handling
- Label cleanup / claim release logic in the shepherd's error/finally path
- `loom-tools/src/loom_tools/shepherd/`

## Impact

- **Correctness**: Successfully merged PRs get their issues put back in the queue
- **Waste**: Another shepherd may attempt to re-implement an already-merged issue
- **Noise**: Exit code 4 on what was actually a successful run

--- a/loom-tools/src/loom_tools/shepherd/cli.py
+++ b/loom-tools/src/loom_tools/shepherd/cli.py
@@ -1995,6 +1995,25 @@ def _cleanup_labels_on_failure(ctx: ShepherdContext, exit_code: int) -> None:
                 pass
         return
 
+    # Check if the PR for this issue is already merged.  If so, the work
+    # is complete and we must NOT revert labels — doing so would re-queue
+    # an already-finished issue.  (See #2515.)
+    try:
+        from loom_tools.shepherd.phases.rebase import _is_pr_merged
+
+        pr_to_check = ctx.pr_number
+        if pr_to_check is None:
+            # Try to find a merged PR by issue number
+            pr_to_check = get_pr_for_issue(issue, state="merged", repo_root=ctx.repo_root)
+        if pr_to_check is not None and _is_pr_merged(pr_to_check, ctx.repo_root):
+            log_info(
+                f"Label cleanup: PR #{pr_to_check} is already merged "
+                f"— not reverting issue #{issue} labels"
+            )
+            return
+    except Exception:
+        pass
+
     # No failure label was set - revert loom:building to loom:issue
     # so the issue returns to the ready pool for retry
     if "loom:building" in current_labels:

--- a/loom-tools/src/loom_tools/shepherd/phases/rebase.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/rebase.py
@@ -10,6 +10,18 @@ from loom_tools.shepherd.labels import add_pr_label, remove_pr_label
 from loom_tools.shepherd.phases.base import BasePhase, PhaseResult
 
 
+def _is_pr_merged(pr_number: int, repo_root: str | None) -> bool:
+    """Check if a PR is already merged via ``gh pr view``."""
+    result = subprocess.run(
+        ["gh", "pr", "view", str(pr_number), "--json", "state", "--jq", ".state"],
+        cwd=repo_root,
+        capture_output=True,
+        text=True,
+        check=False,
+    )
+    return result.stdout.strip() == "MERGED"
+
+
 class RebasePhase(BasePhase):
     """Phase 5.5: Rebase — Update feature branch onto main before merge."""
 
@@ -52,6 +64,15 @@ class RebasePhase(BasePhase):
         if success:
             # Force-push the rebased branch
             if not force_push_branch(cwd=cwd):
+                # Check if the PR was already merged (e.g., by Champion)
+                # before declaring failure — force-push to a merged branch
+                # is expected to fail and is not an error.
+                if ctx.pr_number is not None and _is_pr_merged(
+                    ctx.pr_number, ctx.repo_root
+                ):
+                    return self.success(
+                        f"force-push failed but PR #{ctx.pr_number} is already merged"
+                    )
                 return self.failed(
                     "rebase succeeded but force-push failed",
                     {"reason": "push_failed"},

--- a/loom-tools/tests/shepherd/test_cli.py
+++ b/loom-tools/tests/shepherd/test_cli.py
@@ -3560,6 +3560,47 @@ class TestCleanupLabelsOnFailure:
             repo_root=Path("/fake/repo"),
         )
 
+    @patch("loom_tools.shepherd.cli._cleanup_pr_labels_on_failure")
+    @patch("loom_tools.shepherd.cli.transition_issue_labels")
+    @patch("loom_tools.shepherd.phases.rebase._is_pr_merged", return_value=True)
+    def test_no_revert_when_pr_already_merged(
+        self,
+        mock_merged: MagicMock,
+        mock_transition: MagicMock,
+        mock_pr_cleanup: MagicMock,
+    ) -> None:
+        """Should NOT revert loom:building when the PR is already merged (#2515)."""
+        ctx = self._make_cleanup_ctx(
+            issue_labels={"loom:building", "loom:curated"}
+        )
+        ctx.pr_number = 200
+        _cleanup_labels_on_failure(ctx, ShepherdExitCode.NEEDS_INTERVENTION)
+
+        mock_merged.assert_called_once_with(200, Path("/fake/repo"))
+        mock_transition.assert_not_called()
+
+    @patch("loom_tools.shepherd.cli._cleanup_pr_labels_on_failure")
+    @patch("loom_tools.shepherd.cli.transition_issue_labels")
+    @patch("loom_tools.shepherd.cli.get_pr_for_issue", return_value=300)
+    @patch("loom_tools.shepherd.phases.rebase._is_pr_merged", return_value=True)
+    def test_no_revert_when_merged_pr_found_by_issue(
+        self,
+        mock_merged: MagicMock,
+        mock_get_pr: MagicMock,
+        mock_transition: MagicMock,
+        mock_pr_cleanup: MagicMock,
+    ) -> None:
+        """Should find merged PR by issue number when ctx.pr_number is None (#2515)."""
+        ctx = self._make_cleanup_ctx(
+            issue_labels={"loom:building", "loom:curated"}
+        )
+        ctx.pr_number = None
+        _cleanup_labels_on_failure(ctx, ShepherdExitCode.NEEDS_INTERVENTION)
+
+        mock_get_pr.assert_called_once_with(42, state="merged", repo_root=Path("/fake/repo"))
+        mock_merged.assert_called_once_with(300, Path("/fake/repo"))
+        mock_transition.assert_not_called()
+
 
 class TestMainCleanupIntegration:
     """Test that main() calls _cleanup_labels_on_failure on failure."""

--- a/loom-tools/tests/shepherd/test_rebase.py
+++ b/loom-tools/tests/shepherd/test_rebase.py
@@ -10,7 +10,7 @@ import pytest
 from loom_tools.shepherd.config import ShepherdConfig
 from loom_tools.shepherd.context import ShepherdContext
 from loom_tools.shepherd.phases.base import PhaseStatus
-from loom_tools.shepherd.phases.rebase import RebasePhase
+from loom_tools.shepherd.phases.rebase import RebasePhase, _is_pr_merged
 
 
 @pytest.fixture
@@ -176,11 +176,71 @@ class TestRebasePhase:
                 "loom_tools.shepherd.phases.rebase.force_push_branch",
                 return_value=False,
             ),
+            patch(
+                "loom_tools.shepherd.phases.rebase._is_pr_merged",
+                return_value=False,
+            ),
         ):
             result = phase.run(mock_context)
 
         assert result.status == PhaseStatus.FAILED
         assert "push_failed" in result.data.get("reason", "")
+
+    def test_force_push_failure_but_pr_already_merged(
+        self, mock_context: MagicMock
+    ) -> None:
+        """When force-push fails but PR is already merged, should return SUCCESS."""
+        phase = RebasePhase()
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.rebase.is_branch_behind", return_value=True
+            ),
+            patch(
+                "loom_tools.shepherd.phases.rebase.attempt_rebase",
+                return_value=(True, ""),
+            ),
+            patch(
+                "loom_tools.shepherd.phases.rebase.force_push_branch",
+                return_value=False,
+            ),
+            patch(
+                "loom_tools.shepherd.phases.rebase._is_pr_merged",
+                return_value=True,
+            ),
+        ):
+            result = phase.run(mock_context)
+
+        assert result.status == PhaseStatus.SUCCESS
+        assert "already merged" in result.message
+
+    def test_force_push_failure_no_pr_number_skips_merged_check(
+        self, mock_context: MagicMock
+    ) -> None:
+        """When force-push fails with no PR number, should FAIL without merged check."""
+        mock_context.pr_number = None
+        phase = RebasePhase()
+
+        with (
+            patch(
+                "loom_tools.shepherd.phases.rebase.is_branch_behind", return_value=True
+            ),
+            patch(
+                "loom_tools.shepherd.phases.rebase.attempt_rebase",
+                return_value=(True, ""),
+            ),
+            patch(
+                "loom_tools.shepherd.phases.rebase.force_push_branch",
+                return_value=False,
+            ),
+            patch(
+                "loom_tools.shepherd.phases.rebase._is_pr_merged",
+            ) as mock_merged,
+        ):
+            result = phase.run(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        mock_merged.assert_not_called()
 
     def test_no_pr_number_skips_label_operations(
         self, mock_context: MagicMock
@@ -290,3 +350,31 @@ class TestRebasePhaseValidate:
         mock_context.worktree_path = None
         phase = RebasePhase()
         assert phase.validate(mock_context) is False
+
+
+class TestIsPrMerged:
+    """Tests for the _is_pr_merged helper."""
+
+    def test_returns_true_when_merged(self) -> None:
+        """Should return True when gh reports MERGED."""
+        with patch("subprocess.run") as mock_run:
+            mock_run.return_value = MagicMock(stdout="MERGED\n")
+            assert _is_pr_merged(100, "/fake/repo") is True
+
+    def test_returns_false_when_open(self) -> None:
+        """Should return False when gh reports OPEN."""
+        with patch("subprocess.run") as mock_run:
+            mock_run.return_value = MagicMock(stdout="OPEN\n")
+            assert _is_pr_merged(100, "/fake/repo") is False
+
+    def test_returns_false_when_closed(self) -> None:
+        """Should return False when gh reports CLOSED."""
+        with patch("subprocess.run") as mock_run:
+            mock_run.return_value = MagicMock(stdout="CLOSED\n")
+            assert _is_pr_merged(100, "/fake/repo") is False
+
+    def test_returns_false_on_empty_output(self) -> None:
+        """Should return False when gh returns empty output."""
+        with patch("subprocess.run") as mock_run:
+            mock_run.return_value = MagicMock(stdout="")
+            assert _is_pr_merged(100, "/fake/repo") is False

